### ISSUE TITLE
Performance: Refactor the Collective page to use the newest Next.js data fetching structure

### ISSUE THREAD

**Issue by kewitz** [opencollective/opencollective#6519]
Ultimately, we want to ensure the collective page is server-side rendered as much as possible. We would also like to use Incremental Static Regeneration with on-demand revalidation.

**Comment by Betree** [comment]
I agree, but maybe we can start with a simpler page to have a proof-of-concept.

**Comment by kewitz** [comment]
Closing this issue because I think it is too generic and There were a few things I learned while prototyping this solution:

1. **getStaticProps** is not applicable here. Although this could be used to speed up static pages (home, company, product), I don't think it is worth the effort for the Collective page which is highly customizable. I suggest this effort be directed toward improving our public caching.
2. **getServerSideProps** can be used if we're interested in splitting the CollectivePage query between the part that is frequently updated from the part that contains information that is not often updated if we want to cache them for longer.
  - This function is responsible for generating the JSON file that is used to hydrate pages and we could use this to cache static features of the profile (images, name, etc...) for longer.
  - We could also use this feature to decouple the queries used by logged and unlogged users in a scenario where we want to decrease the DB load by unlogged users and robots.
